src/gval/comparison/pairing_functions.py

```diff
@@ -62,7 +86,7 @@ def _is_not_natural_number(
         return False
 
 
-@nb.vectorize(nopython=True)
+@vectorize(two_param_function_types, nopython=True)
```

Comment:
Are these where the accumulation is occurring?

Reply:
This does not address accumulation but it does address conserving memory when using non 64 bit datatypes.